In [1]:
import pandas as pd
import numpy as np
from sklearn import cross_validation, neighbors, preprocessing, tree, naive_bayes, svm

C:\Users\ASHUU\Anaconda3\lib\site-packages\sklearn\cross_validation.py:44: DeprecationWarning: This module was deprecated in version 0.18 in favor of the model_selection module into which all the refactored classes and functions are moved. Also note that the interface of the new CV iterators are different from that of this module. This module will be removed in 0.20.
  "This module will be removed in 0.20.", DeprecationWarning)


In [2]:
df = pd.read_csv("breast-cancer-wisconsin.data.txt")

In [3]:
df.replace('?', -9999, inplace=True)
df.drop(['id'], 1, inplace=True)

In [4]:
X = np.array(df.drop(['class'], 1))
y = np.array(df['class'])

In [5]:
X_train, X_test, y_train, y_test = cross_validation.train_test_split(X, y, test_size=0.2)

# Training the classifiers and predicting the scores :

In [6]:
# svm classifiers

clf_svm = svm.SVC()
clf_svm.fit(X_train, y_train)
clf_svm.score(X_test, y_test)

0.9642857142857143

In [7]:
#decision tree classifiers
clf_dt = tree.DecisionTreeClassifier()
clf_dt.fit(X_train, y_train)
clf_dt.score(X_test, y_test)

0.94285714285714284

In [8]:
#naive bayes classifiers
clf_nb = naive_bayes.GaussianNB()
clf_nb.fit(X_train, y_train)
clf_nb.score(X_test, y_test)

0.95714285714285718

In [9]:
# k means classifiers
clf_kmeans = neighbors.KNeighborsClassifier()
clf_kmeans.fit(X_train, y_train)
clf_kmeans.score(X_test, y_test)

0.98571428571428577

# Now again retesting the algorithms for validation

In [10]:
# svm classifiers

clf_svm = svm.SVC()
clf_svm.fit(X_train, y_train)
clf_svm.score(X_test, y_test)

0.9642857142857143

In [11]:
#decision tree classifiers
clf_dt = tree.DecisionTreeClassifier()
clf_dt.fit(X_train, y_train)
clf_dt.score(X_test, y_test)

0.94285714285714284

In [12]:
#naive bayes classifiers
clf_nb = naive_bayes.GaussianNB()
clf_nb.fit(X_train, y_train)
clf_nb.score(X_test, y_test)

0.95714285714285718

In [13]:
# k means classifiers
clf_kmeans = neighbors.KNeighborsClassifier()
clf_kmeans.fit(X_train, y_train)
clf_kmeans.score(X_test, y_test)

0.98571428571428577

The result were exactly same


# Now Predicting using an example set

In [14]:
example = np.array([[2,1,2,1,2,1,3,1,1], [6,4,1,1,2,1,2,3,1]])
example = example.reshape(len(example), -1)

In [15]:
#svm
clf_svm.predict(example)

array([2, 2], dtype=int64)

In [16]:
#decision tree
clf_dt.predict(example)

array([2, 2], dtype=int64)

In [17]:
clf_nb.predict(example)

array([2, 2], dtype=int64)

In [18]:
clf_kmeans.predict(example)

array([2, 2], dtype=int64)

# CONCLUSION

# accuracy result:

svm = 0.9642857142857143

decision tree = 0.94285714285714284

naive bayes = 0.95714285714285718

kmeans = 0.98571428571428577

so,
accuracy of
kmeans>svm>naive bayes>decision tree